command/agent/cache/lease_cache.go

```diff
@@ -997,3 +1264,30 @@ func parseCacheClearInput(req *cacheClearRequest) (*cacheClearInput, error) {
 
 	return in, nil
 }
+
+func (c *LeaseCache) hasExpired(currentTime time.Time, index *cachememdb.Index) (bool, error) {
+	reader := bufio.NewReader(bytes.NewReader(index.Response))
+	resp, err := http.ReadResponse(reader, nil)
+	if err != nil {
+		return false, fmt.Errorf("failed to deserialize response: %w", err)
+	}
+	secret, err := api.ParseSecret(resp.Body)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse response as secret: %w", err)
+	}
+
+	elapsed := currentTime.Sub(index.LastRenewed)
+	var leaseDuration int
+	switch index.Type {
+	case cacheboltdb.AuthLeaseType:
+		leaseDuration = secret.Auth.LeaseDuration
+	default:
+		leaseDuration = secret.LeaseDuration
```

Comment:
Can we add a case for handling this by checking for `cacheboltdb.SecretLeaseType` and error on the default case? If somehow we're checking against `TokenType` this may behave unexpectedly.

Reply:
Added in cee5bfb